src/OrchardCore.Themes/TheTheme/Views/MenuItemLink-ContentMenuItem.cshtml

```diff
@@ -1,7 +1,37 @@
+@using Microsoft.AspNetCore.Authorization
+@using OrchardCore.Contents
+
+@inject IContentManager ContentManager
+@inject IAuthorizationService AuthorizationService
 @inject IOptions<AutorouteOptions> AutorouteOptions
+
 @{
     ContentItem contentItem = Model.ContentItem;
     var menuItemPart = contentItem.As<ContentMenuItemPart>();
+    string contentItemId = menuItemPart.ContentItem.Content.ContentMenuItemPart.SelectedContentItem.ContentItemIds[0];
+
+    if (string.IsNullOrEmpty(contentItemId))
```

Comment:
Now I am thinking it might not do what I thought it would. Because the shape will be rendered, but with an empty result. Or is the morphing shape checking the result is not empty before adding a new list item to the menu?

Reply:
We could check if for the part in the menu items and not create the shape to begin with.

Reply:
Check out the changes I made